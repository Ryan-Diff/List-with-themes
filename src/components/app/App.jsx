import React from 'react';
import Theme from '../theme-changer/Theme';

export default function App() {
  return (
    <Theme />
  );
}
