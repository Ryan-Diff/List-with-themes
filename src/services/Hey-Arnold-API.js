export const getCharacters = async(page =1) => {
    const res = await fetch(`https://hey-arnold-api.herokuapp.com/api/v1/characters?page=${page}`);
    const json = await res.json();

    if(!res.ok) throw 'Sorry about it';

    return json.map(character => ({
        id: character._id,
        name: character.name,
        image: character.image
    }));
};
