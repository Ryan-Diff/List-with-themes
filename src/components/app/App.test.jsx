import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { getCharacters } from '../../services/Hey-Arnold-API';

jest.mock('../../services/Hey-Arnold-API');

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    getCharacters.mockResolvedValue([
      {
        'name': 'Ryan',
        'image': 'Ryan.png'
      }
    ]);
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});