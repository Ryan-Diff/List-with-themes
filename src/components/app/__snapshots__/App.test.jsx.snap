// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <header
    class="LightMode"
  >
    <button>
      ヾ⌐■_■ノ♪
    </button>
  </header>
  <ul
    class="CharacterList LightMode"
    data-testid="characters"
  />
</DocumentFragment>
`;
